scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -910,4 +916,66 @@ final boolean isChangeCipherSpecMessageExpected() {
 	protected final void expectChangeCipherSpecMessage() {
 		this.changeCipherSuiteMessageExpected = true;
 	}
+	
+    private static Set<TrustAnchor> getTrustAnchors(X509Certificate[] trustedCertificates) {
+        Set<TrustAnchor> result = new HashSet<>();
+        if (trustedCertificates != null) {
+            for (X509Certificate cert : trustedCertificates) {
+                result.add(new TrustAnchor((X509Certificate) cert, null));
+            }
+        }
+        return result;
+    }
+	
+	/**
+     * Validates the X.509 certificate chain provided by the the peer as part of this message.
+     * 
+     * This method checks
+     * <ol>
+     * <li>that each certificate's issuer DN equals the subject DN of the next certiciate in the chain</li>
+     * <li>that each certificate is currently valid according to its validity period</li>
+     * <li>that the chain is rooted at a trusted CA</li>
+     * </ol>
+     * 
+     * @param trustedCertificates the list of trusted root CAs
+	 * @param message 
+     * 
+     * @throws HandshakeException if any of the checks fails
+     */
+    public void verifyCertificate(X509Certificate[] trustedCertificates, 
+            TrustedRpkStore rpkStore, CertificateMessage message) 
+                    throws HandshakeException {
+        if (message.getCertificateChain() != null) {
```

Comment:
If `rpkStore` is moved to `Handshaker`, the parameter list could be reduced to the`CertificateMessage`.

Reply:
new version pushed